### ISSUE TITLE
Crash if popped before loading completely.

### ISSUE THREAD

**Issue by tdhintz** [SKLn-Rad/Xam.Plugin.Webview#50]
Push a screen that contains this plugin, then pop the view before it finishes loading.  This crashes.

IOS:

System.NullReferenceException Object reference not set to an instance of an object

  at Xam.Plugin.iOS.Extras.FormsWKNavigationDelegate.DidFinishNavigation (WebKit.WKWebView webView, WebKit.WKNavigation navigation) <0x1e9431c + 0x00138> in <14f56fe3ba0541f6adc893777e01d113#dc95e517d089235d9194a0279e54dbb4>:0 
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, System.IntPtr principal, System.IntPtr delegate) <0xb903a8 + 0x00033> in <82393c1969274a079dfefb3c07b134ad#dc95e517d089235d9194a0279e54dbb4>:0 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) <0xb902d0 + 0x000cb> in <82393c1969274a079dfefb3c07b134ad#dc95e517d089235d9194a0279e54dbb4>:0 
  at MyApp.iOS.Application.Main (System.String[] args) <0xfe05c + 0x0002f> in <fcdbc0df473044c9a2ca137e8b33a568#dc95e517d089235d9194a0279e54dbb4>:0 

Android:

System.NullReferenceException Object reference not set to an instance of an object

at Xam.Plugin.Droid.Extras.FormsWebViewClient.OnPageFinished (Android.Webkit.WebView view, System.String url) [0x000a0] in <b655dd2e245b40ac9e425286f3d2b71f>:0 
  at Android.Webkit.WebViewClient.n_OnPageFinished_Landroid_webkit_WebView_Ljava_lang_String_ (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_view, System.IntPtr native_url) [0x00017] in <5005b598e813426186204f9e67a55da5>:0 
  at (wrapper dynamic-method) System.Object:039968b1-fc69-46e5-b4c1-bd85b2a720d9 (intptr,intptr,intptr,intptr)

**Comment by SKLn-Rad** [comment]
Yikes! was sure I was checking the objects in the callbacks before acting on them but thats obviously still not working. Nice find!

Is this a blocker for you? I can get a fix out tomorrow, but it looks like a corner case for most general use purposes.
Thanks for the help.

**Comment by tdhintz** [comment]
Yes.  Testers unfortunately like to flip between views which leads to this crash.  Thank you!

**Comment by Nietoperek** [comment]
Still not fixed?

**Comment by Nico04** [comment]
I have the same issue for month... :(

**Comment by luismts** [comment]
+1
